### DISCUSSION TITLE
Обсуждение механизма принятия решений

### DISCUSSION THREAD

**Discussion by glitchcore** [b4ck5p4c3/0x08.in#361] [General]
На ближайшей встрече предлагается обсудить механизм принятия решений и, соответственно, судьбу чата и канала. Напоминаю, что встречи создавались для очного обсуждения тех вопросов, которые не удается решить онлайн в issues. Предлагаю всем, у кого есть представление о том как мы можем обсуждать и принимать решения, написать это перед встречей, это позволит провести встречу более результативно.

**Comment by glitchcore** [comment]
Напоминаю свою позицию зачем это все нужно:

1. В ХС кто угодно может что угодно делать, если это не нарушает правила, заранее предусмотреть список того, что можно делать а что нельзя — сложно, поэтому был выбран уведомительный принцип принятия решений
2. Некоторые действия затрагивают интересы других людей и могут причинить неприятности, иногда без возможности восстановления либо же восстановление потребует несопоставимых усилий
3. Многие действия требуют поддержи других участников, многие действия могут вызвать поддержку у других участников

По итогу нужно централизованное место, где можно
1. Заявить о своих планах что-то сделать так, чтобы все остальные участники это могли увидеть
2. Получить возражения если действие кому-то мешает
3. Получить помощь или обсудить вопрос

Еще один важный момент: инструмент для обсуждений должен быть удобным и доступным. Практика показала, что обсуждать вопросы в GH готовы не все участники, для многих это вызвает трудности. Создание и обсуждений issue через канал оказалось намного проще.
Однако, динамика обсуждения в чате (возможность и допустимость быстро отправлять множество сообщений) негативно сказывается на качестве обсуждений.

**Reply by heeeyflo** [reply]
Полностью согласен с Андреем.

## Вынос крупных обсуждений в Discussions
Обсуждения в чатах требуют быстрого ответа, пока текущий контекст не съехал в соприкасающийся топик, как это было вчера — обсуждение объективных параметров верстака моментально съехало в обсуждение проблем кооперации, переключив единственный поток обсуждения.

Обсуждение объёмных вопросов в чате требует бо́льших когнитивных усилий из-за:
1. **Необходимости отвечать быстро.** Обсуждения однопоточны, топики быстро перетекают из одного в другой. Нужно торопиться, чтобы успеть высказать мысль пока она ещё релевантна и получит больше внимания.

2. **Спешка провоцирует писать много мелких сообщений.** Сложно потратить десять минут на написание и редактуру сообщения, максимально точно передающего мысль, когда за это время центральная тема обсуждения уже успеет перетечь во что-то другое. 

Предположу, что из-за повышенных когнитивных нагрузок возрастает уровень раздражительности, а следовательно и количество негативных эмоций и оскорблений в обсуждениях.

Формат Discussions идеально решает эту проблему, разделяя всё на чёткие топики и энфорся медленную коммуникацию. Но при этом, чат — это крутой инструмент для быстрой коммуникаций по мелким задачам: уточнить кому какую еду захватить из мака, будет ли сегодня кто-то в спейсе, ко скольки собираемся на киносеанс, или нужно ли кому-то забрать заказ из Робошопа.

## Выпил желтого чата
Считаю, что это очень плохая идея. Мне понятен посыл про элитарность сообщества и свою пещерку, и я согласен что можно больше обсуждений проводить в зелёном чате, разрушая барьер между мемберами и сообществом. Прекрасный пример такой коммуникации есть у [андефспейса](https://undef.club) — один общий чат для всех подряд.

Однако, я не считаю что такой опыт применим к нам:
1. В чате андефспейса сейчас 91 мембер, против 336 в зелёном.
2. В зелёном чате сложилась атмосфера обсуждения всего подряд: мемов, вакцин, литературы, любого оффтопика, кроме самого хакспейса. Стоит ли пытаться перевернуть всё вверх ногами и резко форсить там обсуждение вопросов именно спейса? 

Согласен что нужно поступательно формировать контекст Бэкспейса в зелёном чате, стремиться чтобы хотя бы половина обсуждений была тесно связана с местом, но не считаю что резкий выпил жёлтого чата этому поспособствует.

**Reply by ardenfian** [reply]
Не совсем понятно как формировать контекст именно бэкспейса в зеленом чате. 

Можно выносить на обсуждение в зеленый чат изменения в спейсе, например установку верстака, чтобы начать обсуждение способов установки, материала изготовления и т. д. вместе с теми, кто не является участником.
Вопрос на сколько это интересно тем, кто не имеет статус участника?

Сейчас зеленый чат является некоторой онлайн проекцией деятельности спейса, то есть люди обсуждает свои увлечения. Например выбор микросхем, обсуждение девайсов, безопасности протоколов и всё такое. Напрямую со спейсом не связано, но всё это в рамках деятельности спейса по объединению увлеченных людей. А так как спейс не ограничивает себя определенной тематикой, то не понятно что считать оффтопом. Обсуждение литературы выглядит достаточно уместным.

Важная функция зеленого чата - возможность подглядеть концепцию спейса для потенциалного участника, а так же увидеть, что пространство живет и функционирует и напросится в участники (прямо в чате). Для меня по крайней мере это сработало примерно так.

## Про желтый чат
Не надо называть это "элитарностью", да это членство в клубе, за которое нужно платить как минимум взносами, но наличие этой тусовки это гарантия существования бекспейса именно как хакерспейса, как физического помещения для встреч. А не просто очередного чата. И это нормально, что существует такой порог вхождения, отделяющий готовых вкладываться в дело спейса от просто интересующихся спейсом.
И абосолютно логично иметь свой чат для общения именно с этой группой неравнодушных, да, чат со своей атмосферой на несколько десятков юзеров, разве это плохо? А в чем ещё суть участия? Не только же в доступе к паяльнику, ну. (и да, не иметь желания быть участником полностью нормально, тогда добро пожаловать в _гости_)

Участие в чате не должно быть обязательным конечно, так же как не обязательно регулярное посещение спейса. Поэтому чат не должен становиться местом обсуждения решений относительно самого пространства, для этого есть гитхаб и оффлайн встречи. А вот как это регулировать, неясно, да и всё это может происходить и в зеленом чате, если желтого не будет. Разве что умышленно прекращать общение и референсить собеседника в гитхаб и/или встречу. Но сможем ли?

**Comment by van2048** [comment]
**Относительно выпила желтого чата**

Мне кажется произошли довольно деструктивные действия. Для многих кто по разного рода причинам кто не может регулярно быть в спейсе, желтый част был своего рода пристанищем. Они чувствовали свою причастность к спейсу. Вне зависимости от того как это было на самом деле. По моим скромным подсчетам таких было больше половины. Зеленый чат не является заменой желтому, там сидят чужие а здесь свои. Туда не принято выносить сор на всеобщее обозрение. Не запостишь всякий треш, который парой тут может твориться. Здесь можно писать разные штуки, не боясь ошибиться от того что вероятно пишешь фигню.
Я бы сказал что "спейс как сообщество" - спейс обязан существованию желтого чата. 

Я предлагаю написать бота, который бы выставлял бы "slow mode" на 30 минут в случае если в чате появляются более 50 постов в 10 минут. Я думаю это позволит сильно оздоровить обстановку и сделать споры более осмысленными и содержательными

**Reply by glitchcore** [reply]
Мне нравится эта идея со slow mode, у меня что-то похожее в голове висело.
Ну т.е. мне кажется, что телега доступнее гитхаба, но возможность часто отправлять сообщения ухудшает качество, и иметь автоматическую возможность перевести чат в режим охлаждения при нагреве пердаков может решить проблему.

**Reply by glitchcore** [reply]
На самом деле мы можем просто ввести правило/привычку в случае если участники чувствуют, что пахнет жареным, ставить slow mode.

**Reply by ardenfian** [reply]
Автоматизация выглядит неплохо в данном контексте. У всех может быть разный субъективный порог по включению slow mode, один решит, что пора включать, а кто-то будет парировать, что ещё рано. И снова на этой почве возникнет ругань :)  Если задержку включает бот, то тут кагбы никто и не причем, а если не устраивает порог включения, то можно выставить этот вопрос на оюсуждение на ГХ/встречу.

**Reply by sntxerror** [reply]
если и ставить slow mode (я против) то только на постоянку. опираться на чувства учаскников.. я ни своим ни вашим чувствам не доверяю, сорри. их нужно уметь (это не просто) заметить, понимать, усомниться. в общем большая когнитивная нагрузка помимо того чтобы ходить в клвое место делать свои дела.

**Comment by ghost** [comment]
Я хотел писать что-то похожее на ТЗ. Сперва обозначить желаемое, потом обрисовать воспринимаемое действительное, и пробросить
   несколько стратегий для достижения желаемого.
   Здесь стоит строго понимать, что _желаемое - понятие субъективное_, и
   что для эффективной оргструктуры стоит учитывать желаемое различных
   участников, изменять ее со времением при изменении социальной
   динамики, и в целом поощерять взаимодействие людей с различных
   бэкграундов с различными ценностями для развития всех
   задействованных.

   Здесь еще раз стоит четко указать, что подобная структура несет
   ценность для меня, и все мои восприятия и предложения строго
   мои. Это стоит обозначить чётко, чтобы уменьшить риск свести дискуссию к срачу о
   личных границах.

   Итак, еще раз.
   У меня есть потребность в сообществе, в котором люди с разными
   ценностями сотрудничают друг с другом, и развивают друг друга.
   При этом люди могут приходить и уходить, проекты - начинаться и
   заканчиваться, и сообщество не должно разваливаться всего лишь от
   того, что оно изменяется.

   Между делом стоит упомянуть о том, что слишком поспешное прыгание к
   инструментам и стратегиям приводит к неудовлетворению потребностей
   собщества.
   Стратегия с слишком жесткими временными границами не является
   рабочей, потому что в таком случае будет услышано мнение наиболее
   сильных и вокальных участников.
   Здесь стоит соблюсти баланс между затягиванием переговоров из-за
   неготовности сторон к ним, и тороплению и спешке из-за большей
   силы одной из сторон.
   Все предлагаемые инструменты (например, свод правил по этикету в
   сообществе) адресуют симптомы, а не проблемы, и являются всего лишь
   пластырями для сильных ожогов.

   За все то время, в котором я был в собществе, я наблюдал сущий
   бардак с организацией чего-либо.
   Проекты ведутся и встречи проводятся
   больше на личной инициативе, нежели чем на базе сообщества. Я
   наблюдаю постоянные споры о том, как бы участникам сесть лучше,
   чтобы наконец получился квартет. Отдельные личности держатся за
   инструменты, с которыми они лично привыкли работать.
   Более того, существует организационный лаг из-за того, что общество стало
   крупнее, а инструменты для его организации были приняты тогда,
   когда оно было сильно меньше. Сообществу тесно, но вместо того,
   чтобы выходить из зоны комфорта, оно продолжает наворачивать круги
   вокруг проблем, зубами вцепившись в ставшие неэффективными
   стратегии.
   
   Сообщество состоит из людей. Если люди в сообществе разбиваются в кластеры по
   более специфичным интересам, сообщество начинает состоять из этих
   кластеров. Этот процесс может иметь неограниченную вложенность.
   Если сообщество не аккомодирует этого, кластеры людей с более
   специфичными интересами становятся менее заинтересованными в его
   платформе. Сообщество теряет связность, повышается риск раскола, и
   как следствия, _ослабления сообщества вплоть до его полного распада_.

   Инструмент для организации такого поведения **отсутствует как
   сущность в сообществе**. Более того, из-за вышеупомянутой
   неограниченной вложенности кластеризации подобный инструмент либо
   должен применяться на всех уровнях, и быть тем самым связующим
   звеном сообщества, либо каждый из кластеров, вплоть до отдельных
   индивидов, будут иметь свои инструменты для организации своей
   деятельности, со всеми последствиями.

   Проблема со связностью внешнего и внутреннего миров сообщества
   имеет более глубокий и прикладной характер, чем кажется на первый
   взгляд.
   Эту же проблему решают и участники сообщества, и его отдельные
   кластеры с более специфичными интересами.
   Я утверждаю, что не разделять внешние и внутренние проявления
   активности собщества, его кластеров, и отдельных индивидов в нем -
   верный путь к смятению, отсутствию доверия, и дэдлоку.

   Таким образом, у нас **есть граница между внешним и внутренним
   проявлением определенных групп**. Я не думаю, что кому-то захочется
   участвовать в закупках наборов для пайки для проведения Soldering
   Buddies, кроме организаторов этого события. Это внутренняя
   активность этой группы, и все ее решения по этому событию будут внутренними.
   Тем не менее, обьявление о дате и времени встречи носит чисто
   публичный характер, и является проявлением внешней (публичной)
   активности. У Soldering buddies есть выбор, делать ли это
   самостоятельно, или организовать это на базе сообщества. Если
   сообществво не будет предоставлять платформу, покрывающую нужды
   Soldering Buddies, встречи будут проводиться в другом месте.

   Тот же аргумент применим и к вопросам зеленого/желтого чата. Один
   из них является публичным, в каком-то смысле точкой входа (и
   выхода) в сообщество. Жёлтый чат является приватным для участников
   спейса. Определенная активность в желтом чате просацивается в
   зеленый.
   
   **Моя позиция касательно желтого чата, таким образом, следующая.
   Он жизненно необходим для здорового существования сообщества.**
   Тем не менее, большую часть наблюдаемой активности на данный момент
   в нем стоит приватизировать, чтобы не перегружать ненужными для
   большинства деталями. По сути, я предлагаю постить в желтом чате
   анноунсменты из приватных чатов кластеров с более специфичными
   интересами, или даже личные инициативы отдельных участников.

   Так же стоит ясно обозначить, что в публичном зеленом чате
   непременно будут всплывать темы, не связанные с активностью спейса.
   Подобные события будут совершенно естественны, и не должны
   придушаться на корню. Тем не менее, у сообщества должна быть чёткая
   позиция по соблюдению его внешней границы. Эта позиция - предмет
   дальнейших обсуждений, и скорее всего, эти обсуждения не должны
   затрагивать всех участников, каким бы странным это не казалось.

   Здесь существует важный момент с организацией связности между
   внешней и внутренней стороной сообщества.
   На данный момент таким инструментом является live-канал, который
   работает в направлении private-public. 
   В направлении public-private организованная связность пока
   отсутствует, для ее наличия был telegram-бот, который на момент
   написания данного поста сломан.

   Сообщество должно поощерять подобную связность внешних и внутренних
   сторон для своих подгрупп и людей в нем. На данный момент у нас
   нету хорошего примера, и каждый лепит кто во что горазд.
   Еще раз, **основная идея в том, что жёлтый чат сообщества является
   (ну или должен являться) "зеленым чатом" для кластеров по
   специфичным интересам на его платформе**.

   У меня есть идеи касательно того, как можно организовать подобную связность. Нам
   нужен человек (или группа) что снаружи границы, что внутри. Наличие
   пары агентов с налаженной коммуникацией по кастомному протоколу
   будет гарантировать поток информации церез границу в обоих направлениях.

   Мы не можем навязывать механизм принятия решений для подгрупп на
   базе сообщества. Единственное, что мы можем сделать -
   спроектировать и реализовать благоприятные условия для
   плодотворного и конструктивного взаимодействия вышеупомянутых
   подгрупп. Если эти условия окажутся достаточно рабочими, подгруппы
   будут брать их за пример и основу внутренней коммуникации.

   Таким образом, **принимать решение по поводу механизма принятия
   решений и привязываться к инструментам будет слишком рано**. Я
   предлагаю начать с прорабатывания вопросов внешних и внутренних
   сторон у границы сообщества, его подгрупп, и его участников, и
   наиболее часто встречающихся процессов при их взаимодействии.
   У нас есть хорошая заготовка для таких переговоров, @z4fd принес
   ссылку (https://www.noisebridge.net/wiki/FractalWorkingGroup). 
   Продолжать предлагаю с неё.

**Reply by sntxerror** [reply]
Поддерживаю идею формирования процесс рабочиъх груп, но считаю что адаптация доки от noisebridge может идти паралельно с определением процесса принятия решений.

Проще гворя не считаю отсутствие рабочих групп блокирующим.

**Reply by kitnerboy** [reply]
Прошу прощения за возможный оффтоп, но не могу не спросить: @adjofun форматирование твоего комментария - результат работы гитхаба, или это последствия копирования текста из другого источника?
Конкретно меня интересуют переносы строк на середине строки.
![image](https://user-images.githubusercontent.com/14994941/108994892-43d1ea80-7694-11eb-9249-29aa3a365e27.png)
Вот так это вижу я.

**Reply by ghost** [reply]
У emacs есть некий auto-fill-mode, который держит текст в пределах 80ти символов в строке.
К сожалению, я торопился, и редактировал текст переносами самостоятельно, после вставки важных пояснений и примеров.

**Reply by kitnerboy** [reply]
Постарайся, пожалуйста, избежать такого в дальнейшем - длинный текст в таком виде становится трудным для чтения на большой дисплее.
На маленьком выглядит так же плохо, как неформатированный, так что всё в порядке. :)

**Reply by ghost** [reply]
Не обещаю :ъ

**Comment by sntxerror** [comment]
Не знаю как у остальных, но у меня сложилась цельная картина фреймворка обсуждений и принятия решений. Постараюсь описать здесь. 

**Важно постараться организовать мехнизм обсужения и принятия решений на уже существующих инструментах.** 

У меня есть стремление не создавать задачи для участников на ровном месте. По возможности , для достижения целей максимально использоать общепринятые и готовые решения, адаптировать процессы под них. Стараться обойтись как можно меньшим количеством универсальных инструментов. Тоесть пока это возможно, пока это не остро необходимо:

- не проектировать собственные сложные системы, которые потом сложно и некому будет поддерживать
- не заводить новые сущности связанные со спейсом в готовых системах

Итого, в контексте топика я предлагаю использовать два уже существующих, принятых инструмента - гитхаб и телеграм.

## Коллективно принимать решения и вести здоровую коммуникацию не легко.

Некоторые уже успели заметить (ктото еще полтора года назад), что обсуждение задач и решений среди участников бывает не простым. Будь то обсуждение голосом на встрече участников и вне неё, в issues на гитхабе, в пулл-реквесте, в чате участников, где угодно.  Обсуждать и принимать решения коллективно - трудно. Так же трудно как и вести здоровую коммуникацию в целом.

Это ни в коем случае не вина участников, это распространенное свойство индивидуального развития психики, и восприятия людей. Кроме тго, без сомнения, среди участников есть ребята кому особенно сложно общаться, в виду например социальной тревоги или других состояний, которые невозможно коллективно учесть при кждом общении.

Эффекивная, простая, здоровая  коммуникация этовнимательность к себе, к другим, к своим чувствам, и к чувствам других, это труд над собой, труд в коллективе, для этого нужны силы, ресурсы, желание и умение. Мы не можем требовать этих качеств от всех, это было бы слишком идеалистично, мы вынуждены принять это как есть и помогать себе.

**Выбранные нами инструменты должны помогать облегчать коммуникацию, а не усложнять её.**

Два инструмента, два подхода к коммуникации.

Телеграм - моментальный, с богатым ассортиментом способов выражения эмоций (стикеры, эмодзи, частота отправки сообщений и т.д.). Отлично подходит когда нужно оперативно связаться, бстро передать информацию.

1. Чаты в телеграме благодаря своей моментальной природе позволяют общаться очень быстро, настолько быстро, что некоторые говорящие могут не успеть осознать свои чувства и эмоции, не успевают не просто заметить их, но и в том числе, в них усомниться, что критически важно для ненасильственного конструктивного общения. В конечном счете моментальность усложняет коммуникацию для всех собеседников в чате.

_Еще не написал свой ответ, уже кто-то ответил так, что поменял твою позицию, волна эмоций захватывет от еще старого сообщения, они уже перетекают в новое, нужно успеть ответить пока тема актуальна, пара сообщений и суть уже потеряется..._ 

2. Помимо скорости общения, есть и другое свойство - это очень простая, легкая потеря контекста обсуждения. При этом ход диалога и достигнутый прогресс в обсуждении кране трудно отследить.

_Обсуждали вопрос1, кто-то пришел спустя полчаса горячего спора, ответил эмоционально на первое сообщение, другой подметил эмоцию, стали обсуждать вопрос эмоций, обсуждали эмоцию всопмнили вопрос2, обсуждаем все вперемешку, ни к чему не приходим, важный вопрос не решен, получен коллективный стресс, в остатке чувство вины и копящаяся обида..._

3. Другие важные факторы - это доступность чата всегда под рукой и, как следствие, повышенное влияние внешней среды, обстановки собеседников на обусждения и решения. Казалось бы, доступность выглядит как преймущество, но на деле, повсеместность телеграма влечет за собой внезапность обуждений, их спонтанность и позволяет с легкостью привносить нежелательное влияние внешней среды в общение.

_Пошел в магазин, наступил в лужу, сердито полистываешь экран в долгой очереди, какраз когда в чате внезапно почему-то происходило обсуждение важного решения. Будучи уже неспокойным, пишешь свой ответ менее вежливо чем обычно, из-за отсутствия интонации в тексте тебя читают еще грубее, в итоге обсуждение подогрето на ровном месте._

4. За чатом невозможно постоянно следить, а если принимаемые решения важны для вас, то делать это приходиться. Если человек прямо сейчас не может обуждать вопрос в чате, скорее всего всё обсудят без него и он даже не узнает об этом. При этом позиция, аргументы, знания пропустившего могли бы помочь эффективнее решить вопрос. 

Я уверен можно найти и другие аспекты негативно влияющие на диалоги в телеграме, но надеюсь описанных мной будет достаточно чтобы проилюстрировать мысль. 

Резюмируя телеграм:

- моментальность = невнимательность + эмоции
- простая потеря контекста = низакая эффективность + неотслеживаемость решений
- повсеместность = внезапность + повышенное влияние внешней среды
- невозможность постоянного пребывания = потеря важных мнений, аргументов

**Мы априори не можем полностью исключить эти факторы, но нам безусловно под силу снизить их влияние. Для этого необходимо пересмотреть наши подходы к использованию инструментов.**

Обсуждения на гитхабе позволяют все эти факторы если не исключить, то как минимум существеноо снизить их воздействие. Представьте сами:

_Когда есть время, возможность, вы открываете ноутбук, скорее всего вы не застряли в пробке в этот момент и вокруг достаточно подходящая среда. Вы зашли в обсужденмя, видите список тем. Наиболее актуальные - сверху. В них прослеживается история, авторы сообщений не теряют контекст. Общение вежливое, взвешенное, аргументированное. Ваш прошлый комментарий прочли и откоментироали, он мог повлиять на предлагаемое решение. Вы прочли позиции других участников, вдумчиво описали свое предложение. Убрали пару горячих фраз перед отправкой, добавили разметку. Прочли еще раз и отправили._ 

Подобная динамика выглядит заметно более здоровой, а диалог в таком ритме ведется существенно более конструктивный.

Все же, помимо растянутого процесса, участникам важно иметь возможность моментального оповещения о происходящем. Для этого достаточно иметь закрытый	 телеграм-канал участников, с возможностью публикации. Место для любой информации важной другим участникам.

_События на гитхабе (новая тема, PR или issue) могут попадать в канал автоматически, сообщения важные для других участников можно всегда отправить вручную. Привлечь внимание к проекту, к своему комментарию в обсуждении или сообщить о новой схеме работы СКУД..._

_Возникла идея - создаем обсуждение (линк в канал), нашли ошибку на сайте - поправлю в PR (линк в канал), забыли шапку в спэйсе - пишем на всех, нашедшие откликнутся в личку_  

**При этом никто не в праве запретить собирать чаты, общаться в них, шутить, отдыхать.**

Желтый чат сам по себе ничем в сути своей не отличается от других. Отличаются те обычаи, которые в нем для многих из нас приняты. Иначе говоря, привычки общения, среди которых довольно много вредительных.

**Я считаю что желтый чат изжил себя в том виде в котором он нам стал привычен. Общение в нем очень давно перестало быть конструктивным из-за обилия плохих привычек общения**

Считаю так же что нам необходимо отказаться от него совсем.

Конечно, можно собрать другой чат, где не будут обсуждаться реешния и важные общие вопросы спейса. Но привычки в нем должны быть сформированы так, что участники будут сами модерировать и поддерживать друг друга на предмет вынесения темы в discussion. Тем не менее считаю это почти невозможным (так уже было в желтом - не вышло).

Кроме того считаю что пребывание в нем не означает причастности к спейсу, эту функцию на себя мог бы взять закрытый канал.

Подытоживая, добавлю, что **текущая ситуация на мой взгляд блокирующая, поэтому любое из выбираеиых в итоге нами решений стоит оценить простым вопросом "Мы сможем использовать его уже сейчас?"**

## Предложения

- отказаться от необходимости общего чата всех участников спэйса (т.н. желтый чат) и пребывания в нем
- закрепить за участниками подписку на приватный информационный канал 5UDOER5
- обсуждения вопросов, решений касаемо спэйса вести **только** в github discussions
- в виде issues заводить только баги, ошибки, фичи сайта правки документации и кода
- принять за правило публикацию сообщений в канал участников (ссылки на обсуждения, issues, PR, важные для всех объявления)
- фиксировать решения принятые на встречах участников как новый disscussion
- заняться процессом организации рабочих групп

**Comment by glitchcore** [comment]
Проблематика:
1. Есть потребность обсуждать важные вопросы
2. Сейчас обсуждение важных вопросов происходит в чате.
 -- Обсуждения эмоциональны.
 -- Сложно обработать неструктурированный поток информации
 -- Легкая потеря контекста обсуждения.
В итоге обсуждение важных вопросов происходит неэффективно и вызывает желание его проигнорировать.

**Reply by sntxerror** [reply]
> -    моментальность = невнимательность + эмоции
> -   простая потеря контекста = низакая эффективность + неотслеживаемость решений
> -  повсеместность = внезапность + повышенное влияние внешней среды
> - невозможность постоянного пребывания = потеря важных мнений, аргументов

**Reply by glitchcore** [reply]
Согласен, все так

**Comment by glitchcore** [comment]
Чо придумали:

1. Создаем/поддерживаем канал с issue в телеграме
2. Делаем приватный чат, readonly для всех (технически/ботом/правилами), привязываем его к комментам канала. (Есть идея в будущем включить для него slow-mode 30s).
3. Создаем/поддерживаем discuss обсуждение на GH, синхронизируем с каналом/обсуждениями
4. Делаем/поддерживаем отдельный чат для оффтопика
5. Добавляем бота, который предупреждает о горении чата, рекомендует создать issue (в случае с флудилкой) или переехать в discuss (в случае с комментариями к issue), просто выдохнуть, включает slow-mode

**Reply by glitchcore** [reply]
В связи с п. 1 — переименовать существующий канал в `B4CKSP4CE | I55UE5`. @sntxerror сможешь сделать?
В связи с п. 2 — создать чат, настроить приватность, привязать к каналу. @van2048. Мб @z4fd добавит в бота фичу удаления сообщений, не привязанных к тредам
В связи с п. 3 — @imcatwhocode высказал предложение заняться синхронизацией
В связи с п. 4 — переименовать существующий Вольчек-чат в `B4CKSP4CE | ФлУд`. @sntxerror сможешь сделать?
В связи с п. 5 — @z4fd высказывал желание сделать бота

**Reply by rusdacent** [reply]
>  В связи с п. 2 — создать чат, настроить приватность, привязать к каналу. @van2048. Мб @z4fd добавит в бота фичу удаления сообщений, не привязанных к тредам

В телеге есть встроенный slow mode. Можно попробовать его сразу включить в этот канал, например, на 30 секунд.
![slow_mode](https://user-images.githubusercontent.com/57439765/108909866-29692400-7636-11eb-8b12-26f2cbac9c21.png)

Обсуждаемо, естественно, но просто что б донести мысль об этом.

**Reply by glitchcore** [reply]
Так в тексте предложения так и написано «Есть идея в будущем включить для него slow-mode 30s». Мне кажется, это смотивирует участников более вдумчиво писать обсуждения и приблизит этот канал к гитхабу по стилю постов

**Reply by glitchcore** [reply]
Еще один, кмк важный момент: если будет настроена трансляция комментариев в публичный GH, это может добавить ответственности перед авторами комментариев

**Reply by sntxerror** [reply]
сегодня телега выкатила обновление с автоудалением сообщений в  групповом чате по таймеру. предлагаю установить такой таймер в чате обсуждений канала на 24 часа. Тогда точно, все обсуждения тебующие фиксации будут выноситься в дискач и контекст в чате обсуждений не будет размываться. Дополнительная мотивация, кмк.

https://telegram.org/blog/autodelete-inv2/ru

**Reply by kitnerboy** [reply]
> создать чат, настроить приватность, привязать к каналу.

А какой функционал будет выполнять бот?
Чем ограничен вариант, который временно создавал я: к каналу привязывается приватный чат, в котором никто не состоит кроме одного админа (тут, согласен, есть некоторая социальная уязвимость).

**Reply by sntxerror** [reply]
> А какой функционал будет выполнять бот?

бот будет заниматься модерацией беседы и просить выносить интенсивные топики в дискач\влючать слоумод

**Reply by glitchcore** [reply]
> Тогда точно, все обсуждения тебующие фиксации будут выноситься в дискач, а все моментальные будут доступны момнетально

Меня смущает что они из телеги совсем пропадают, т.е. чтобы увидеть обсуждение придется обязательно идти в GH. Если будет полноценная синхронизация, необходимости в этой фиче не вижу. Если мы делаем синхронизацию руками —да, автоудаление может быть мотивацией не забывать перетаскивать обсуждение.

**Reply by glitchcore** [reply]
> А какой функционал будет выполнять бот?
> Чем ограничен вариант, который временно создавал я: к каналу привязывается приватный чат, в котором никто не состоит кроме одного админа (тут, согласен, есть некоторая социальная уязвимость).

Короче проблема в том, что в телеге нельзя подписаться на обсуждения поста. Т.е. я задал вопрос и приходится поллить комментарии к нему. @rusdacent предложил вариант, когда на приватный канал с обсуждениями все подписаны, тогда ты видишь что туда прилетело новое сообщение и смотришь, твое ли обсуждение. Но появляется соблазн написать прямо в чат. Для этого было предложено добавить бота, который будет в т.ч. вычищать сообщения, не являющиеся частью обсуждения поста.

**Reply by glitchcore** [reply]
На самом деле, если мы сделаем синхронизацию с GH, то можно будет подписываться на обновления disucssion-топиков средствами гитхаба и не залезать в приватный чат.

**Reply by sntxerror** [reply]
>  Если мы делаем синхронизацию руками

пока никакой синхронизации нет, предлагаю все таки использовать автоудаление, чтобы не размывать кнтекст и поддерживать мотивацию идти в дискашн

**Reply by kitnerboy** [reply]
Хотел бы обозначить потенциальную опасность в обсуждениях по крайней мере при наших настройках приватности:
Мы можем редактировать чужие комментарии.
Это можно пронаблюдать на примере комментария @adjofun, где я убрал перенос из первой строки. 
Полагаю, никто не мешает мне и лишние мнения убрать.
Для меня не очень очевидны настройки приватности в репе, вся их ветвистость и сложность, потому пока не могу предложить решения по этой проблеме.

**Reply by ghost** [reply]
Никаких уведомлений об этом мне на почту не пришло, хочу заметить.

**Reply by kitnerboy** [reply]
![image](https://user-images.githubusercontent.com/14994941/108998588-0b80db00-7699-11eb-855e-bb3eb141bc33.png)
Это единственная улика, доказывающая мою вину.

**Reply by z4fd** [reply]
> В связи с п. 2 — создать чат, настроить приватность, привязать к каналу. @van2048. Мб @z4fd добавит в бота фичу удаления сообщений, не привязанных к тредам

Сделал POC, сегодня отлажу и можно тестировать ITW

**Comment by glitchcore** [comment]
Я посмотрел за нашим поведением в оффтоп чате. Мне кажется, что название чата и _**шрамы на наших сердцах**_ располагают к тому чтобы отправлять всех делать треды и обсуждать их. Кроме того, если когнитивная нагрузка для перехода будет не высокой (типа «писать не в одну вкладку телеграма а в другую»), переход будет получаться и войдет в привычку.
